vispy/visuals/bar.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+
+import numpy as np
+from .mesh import MeshVisual
+
+
+class BarVisual(MeshVisual):
+    """Visual that calculates and displays a bar graph
+
+    Parameters
+    ----------
+    data : array-like
+        1st column is the height, optional 2nd column is bottom
+    height : array-like
+        Height of the bars
+    bottom : array-like
+        Bottom of the bars
+    width : int | float
+        Width of all bars
+    shift : int | float
+        Shift of all bars along the x-axis
+    color : instance of Color
+        Color of the bar.
+    orientation : {'h', 'v'}
+        Orientation of the histogram - 'h' is default
+    """
+
+    def __init__(self, height, bottom=None, width=0.8, shift=0.0, color='w', orientation='h'):
+        if bottom is None:
+            bottom = np.zeros(height.shape[0])
+        
+        rr, tris = calc_vertices(height, bottom, width, shift, orientation)
+
+        MeshVisual.__init__(self, rr, tris, color=color)
+
+    def update_data(self, height, bottom=None, width=0.8, shift=0.0, color='w'):
+        if bottom is None:
+            bottom = np.zeros(height.shape[0])
+        rr, tris = calc_vertices(height, bottom, width, shift=shift)
+
+        MeshVisual.set_data(self, rr, tris, color=color)
+
+
+def calc_vertices(height, bottom, width, shift, orientation):
+
+    y_position = np.zeros((height.shape[0], 2))
+    
+    y_position[:, 0] = height
+    y_position[:, 1] = bottom
+
+    y_position = y_position.flatten().repeat(2)
+
+    stack_n_x2 = np.arange(0, height.shape[0], dtype=int) 
+    stack_n_x2 = np.expand_dims(stack_n_x2, axis=1)
+    stack_n_x2 = np.repeat(stack_n_x2, 1, axis=1)
+    stack_n_x2 = np.repeat(stack_n_x2, 2, axis=0)
+
+    vertices = np.array([
+        [-width/2 + shift, 1, 0],
+        [width/2 + shift, 1, 0],
+        [width/2 + shift, 0, 0],
+        [-width/2 + shift, 0, 0]])
```

Comment:
What's the point of the `0` at index 2? I'm having a hard time following the flow.

Reply:
Well I thought that vertices would have to be x,y,z and the resulting vertices for the mesh should be in the shape (n,3), since its a 2d plot the z axis is zero. I'm not married to it if you want to change it it would be fine with me

Reply:
Ok, I see why I was confused: this resulted in a deprecation warning from numpy due to a staggered sequence in my testing, because I was passing an array to `width` rather than a single value... My bad :P

Reply:
But yeah, I think you could remove the zero there and it would be fine, meshes work well with 2D data and we do it in several other places as well.